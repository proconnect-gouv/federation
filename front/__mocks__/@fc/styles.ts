import { StylesProviderProps } from '@fc/styles';
import React, { ReactElement } from 'react';

export const useStylesQuery = jest.fn();

export const useStylesContext = jest.fn();

export const useStylesVariables = jest.fn(() => []);

export const StylesProvider = jest.fn(
  ({ children }: StylesProviderProps) => children as ReactElement,
);

export const StylesContext = React.createContext(expect.any(Object));

export enum Strings {
  DOUBLE_DASH = '--',
}
