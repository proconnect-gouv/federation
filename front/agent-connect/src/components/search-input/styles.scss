@import './../../styles/_colors.scss';
@import './../../styles/_variables.scss';

#identity-provider-search #fi-search-form {
  label[for='fi-search-term'] {
    color: $charcoal-grey;
    display: block;
    font-size: 18px;
    height: auto;
    line-height: 1.8rem;
    padding-bottom: $spacer;
    text-align: center;
  }

  input[name='fi-search-term'] {
    border: 1px solid #d9d9d9;
    border-radius: 0.25rem;
    float: none;
    height: 60px;
  }

  button[type='submit'] {
    display: block;
    height: 60px;
    margin-left: 0;
    margin-top: $spacer;
    width: 100%;

    &:hover {
      text-decoration: underline;
    }
  }

  .ant-input-group {
    display: flex;
    flex-direction: column;
  }

  .ant-input-group-addon {
    background: transparent;
    display: block;
    flex: 1;
    width: 100%;
  }
}

@media (min-width: $breakpoint-mobile) {
  #identity-provider-search #fi-search-form {
    label[for='fi-search-term'] {
      text-align: center;
    }

    button[type='submit'] {
      margin-top: 0;
      width: 100%;
    }

    .ant-input-group {
      flex-direction: row;
    }

    .ant-input-group-addon {
      flex: 0;
      padding-left: $spacer;
      width: 210px;
    }
  }
}
