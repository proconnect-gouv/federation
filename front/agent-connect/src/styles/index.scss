// ----------------- VARIABLES
@import '_variables.scss';
@import '_colors.scss';

// ----------------- VENDORS
@import '~antd/dist/antd.css';
@import '~bootstrap/scss/bootstrap';
@import '~sass-rem';

// ----------------- CUSTOM
@import './helpers';
@import './vendors';
@import './fonts/montserrat';

*,
input,
textarea {
  font-family: 'Montserrat', sans-serif;
}

body,
html,
#root {
  // @include rem-baseline;

  color: $charcoal-grey;
  height: 100%;
  min-height: 100%;
}

a,
.btn-link {
  color: $charcoal-grey;

  &:hover {
    color: $primary;
  }
}

.unordered-list {
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    display: flex;
    flex-direction: row;
    padding: 2px 5px;

    &::before {
      content: 'â€¢';
      font-size: 36px;
      line-height: 0.85;
      margin-right: 12px;
    }
  }
}
