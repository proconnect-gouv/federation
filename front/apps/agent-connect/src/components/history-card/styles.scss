@import '@fc/dsfr/styles.scss';
@import '@fc/dsfr/core/funcs/pixel-to-rem';

.identity-provider-card {
  max-width: 100%;

  .title {
    line-height: 1;
  }

  & .disabled {
    background: $blue-agentconnect;
    opacity: 0.5;
  }

  .card-wrapper {
    height: calc(100% - 24px);
    height: calc(100% - 1.5rem);
    border-bottom: 4px $blue-agentconnect solid;
  }

  .form-title {
    color: $g700;
    font-family: 'Montserrat', sans-serif;
  }

  .form-button {
    color: $blue-agentconnect;
    text-decoration: none;
  }

  @media screen and (min-width: 768px) {
    @include rem(max-width, 343px);
    width: 33.3%;
  }
}
