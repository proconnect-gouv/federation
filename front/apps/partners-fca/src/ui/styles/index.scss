body,
html,
#root {
  height: 100%;
  min-height: 100%;
}

#root {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

#page-container {
  max-width: 600px;
}
