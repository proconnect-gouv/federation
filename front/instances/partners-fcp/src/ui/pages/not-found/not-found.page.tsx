export const NotFoundPage = () => (
  <div className="fr-container text-center fr-mt-8w">
    <h1 className="text-center">404 - Not Found</h1>
  </div>
);
