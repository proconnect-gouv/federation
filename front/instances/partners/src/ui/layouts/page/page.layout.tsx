import React from 'react';
import { Outlet } from 'react-router-dom';

export const PageLayout = React.memo(() => (
  <main className="fr-container fr-py-8v">
    <div className="fr-grid-row fr-grid-row--center">
      <Outlet />
    </div>
  </main>
));

PageLayout.displayName = 'PageLayout';
