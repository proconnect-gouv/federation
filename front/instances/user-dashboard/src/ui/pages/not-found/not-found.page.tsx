import classnames from 'classnames';
import React from 'react';

import { useStylesQuery, useStylesVariables } from '@fc/styles';

export const NotFoundPage = React.memo(() => {
  const [breakpointLg] = useStylesVariables(['breakpoint-lg']) as unknown as string;
  const gtDesktop = useStylesQuery({ minWidth: breakpointLg });

  return (
    <div
      className={classnames('fr-m-auto fr-px-2w text-center', {
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-5w': !gtDesktop,
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-8w': gtDesktop,
      })}
      id="page-container">
      <h1 className="text-center">404 - Not Found</h1>
    </div>
  );
});

NotFoundPage.displayName = 'NotFoundPage';
