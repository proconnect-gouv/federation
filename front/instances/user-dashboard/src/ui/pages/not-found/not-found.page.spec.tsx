import { render } from '@testing-library/react';

import { useStylesQuery, useStylesVariables } from '@fc/styles';

import { NotFoundPage } from './not-found.page';

describe('NotFoundPage', () => {
  beforeEach(() => {
    // Given
    jest.mocked(useStylesVariables).mockReturnValue([expect.any(String)]);
  });

  it('should match the snapshot, when view is lower than a tablet view', () => {
    // Given
    jest.mocked(useStylesQuery).mockReturnValueOnce(false);

    // When
    const { container } = render(<NotFoundPage />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, when view is greater or equal than a tablet view', () => {
    // Given
    jest.mocked(useStylesQuery).mockReturnValueOnce(true);

    // When
    const { container } = render(<NotFoundPage />);

    // Then
    expect(container).toMatchSnapshot();
  });
});
