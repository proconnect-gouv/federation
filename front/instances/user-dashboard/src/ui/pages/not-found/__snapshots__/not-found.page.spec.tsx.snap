// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should match the snapshot, when view is greater or equal than a tablet view 1`] = `
<div>
  <div
    class="fr-m-auto fr-px-2w text-center fr-mt-8w"
    id="page-container"
  >
    <h1
      class="text-center"
    >
      404 - Not Found
    </h1>
  </div>
</div>
`;

exports[`NotFoundPage should match the snapshot, when view is lower than a tablet view 1`] = `
<div>
  <div
    class="fr-m-auto fr-px-2w text-center fr-mt-5w"
    id="page-container"
  >
    <h1
      class="text-center"
    >
      404 - Not Found
    </h1>
  </div>
</div>
`;
